### PR TITLE
[SafeSnap] Multisend configuration

### PR DESCRIPTION
**Pull request by carlosfebres**
**:warning: [THIS PR](https://github.com/snapshot-labs/snapshot-plugins/pull/62) IN THE [SNAPSHOT-PLUGINS REPO](https://github.com/snapshot-labs/snapshot-plugins) NEEDS TO BE MERGED FIRST :warning:**

### Summary
This allows a safe to use their desired  [MultiSend contract](https://github.com/gnosis/safe-contracts/blob/da66b45ec87d2fb6da7dfd837b29eacdb9a604c5/contracts/libraries/MultiSend.sol) to batch transaction.

Example of a SafeSnap configuration using a specific MultiSend contract:
```JSON
{
  "safes":[
    {
      "network":"4",
      "realityAddress":"0x5CC00E150CbFB64039Bdd076911ff0a81180F1b3",
      "multiSendAddress":"0xA238CBeb142c10Ef7Ad8442C6D1f9E89e07e7761"
    }
  ]
}
```

### CODE REVIEW

**Comment by carlosfebres** [comment]
Build is failing because [this PR](https://github.com/snapshot-labs/snapshot-plugins/pull/62) in the [snapshot-plugins repo](https://github.com/snapshot-labs/snapshot-plugins) needs to be merged first

File: src/components/Ui/CollapsibleContent.vue [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import chevronIcon from '@/assets/icons/chevron.svg';
+
+export default {
+  props: ['open', 'title', 'number', 'hideRemove', 'showArrow', 'borderless'],
+  emits: ['toggle'],
+  data() {
+    return { chevronIcon };
+  }
+};
+</script>
```
Comment (mktcode) [inline src/components/Ui/CollapsibleContent.vue:11]: I'd like to encourage you to use composition API. Would be much shorter.

```suggestion
<script setup>
import chevronIcon from '@/assets/icons/chevron.svg';

defineProps(['open', 'title', 'number', 'hideRemove', 'showArrow', 'borderless']);
defineEmits(['toggle']);
</script>
```

**Review by mktcode — CHANGES_REQUESTED** [review]
Tbh I don't like the fact that we're spreading the logic so much. E.g. why does [this](https://github.com/snapshot-labs/snapshot/pull/1490/files#diff-cf15551aead4243571d2b95721dd64b732b32ecff874318397d932c22b293ab6) need to be in the frontend? We are currently hardcoding so much plugin stuff in places that have nothing to do with it. Could you maybe double check and try to move as much stuff as possible to the plugins repo?

**Comment by carlosfebres** [comment]
> Tbh I don't like the fact that we're spreading the logic so much. E.g. why does [this](https://github.com/snapshot-labs/snapshot/pull/1490/files#diff-cf15551aead4243571d2b95721dd64b732b32ecff874318397d932c22b293ab6) need to be in the frontend? We are currently hardcoding so much plugin stuff in places that have nothing to do with it. Could you maybe double check and try to move as much stuff as possible to the plugins repo?

@mktcode I agree with you and that's something I wanted to do a while ago and started moving things to the `snapshot-plugins` repo in this PR, I moved a bunch of functions, let me know if you find something else that can be moved.

**Review by mktcode — APPROVED** [review]
Tested. Looks good! See comments here: https://github.com/snapshot-labs/snapshot-plugins/pull/62

**Review by mktcode — APPROVED** [review]

**Comment by carlosfebres** [comment]
@mktcode re-running checks here should merge this 🚀

**Review by mktcode — APPROVED** [review]

**Comment by mktcode** [comment]
`'formatBatchTransaction' is not exported by snapshot-plugins/src/plugins/safeSnap/index.ts, imported by src/components/Plugin/SafeSnap/HandleOutcome.vue`

I can handle that in a bit.... just need to eat something first. :D

**Comment by mktcode** [comment]
Ok... It's just the plugins submodule that needs to be updated. But...

I don't really get how it works then because I can't find that export of `formatBatchTransaction`.

In general this is a bit confusing and obfuscating the true origin of things.

https://github.com/snapshot-labs/snapshot-plugins/blob/70c7c07497c8af7dcee953b583b17f709fba1457/src/plugins/safeSnap/index.ts#L28-L39

Didn't we change that?

**Pull request merged**